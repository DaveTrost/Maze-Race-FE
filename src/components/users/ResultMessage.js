import React from 'react';
import PropTypes from 'prop-types';

const ResultMessage = ({ handleSubmit, name, winner }) => {

  return (
    <>
      <h1>
        {name === winner ? 'You win!' : 'You Lost'}
      </h1>
      <button onClick={handleSubmit}>Play Again?</button>
    </>
  );
};

ResultMessage.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  name: PropTypes.string,
  winner: PropTypes.string
};

export default ResultMessage;
