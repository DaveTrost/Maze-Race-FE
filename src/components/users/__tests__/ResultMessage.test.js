import React from 'react';
import { shallow } from 'enzyme';
import ResultMessage from '../ResultMessage.js';

describe('ResultMessage component', () => {
  it('renders ResultMessage', () => {
    const wrapper = shallow(<ResultMessage />);
    expect(wrapper).toMatchSnapshot();
  });
});
