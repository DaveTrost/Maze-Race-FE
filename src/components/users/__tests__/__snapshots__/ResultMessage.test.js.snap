// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultMessage component renders ResultMessage 1`] = `
<div
  className="ResultsMessage"
>
  <h1>
    You win!
  </h1>
  <button
    onClick={[Function]}
  >
    Play Again?
  </button>
  <Link
    to="/about"
  >
    <h3>
      About the Developers
    </h3>
  </Link>
</div>
`;
